torchtext/datasets/text_classification.py

```diff
@@ -0,0 +1,484 @@
+import os
+import re
+import torch
+from ..utils import download_and_extract_archive, makedir_exist_ok
+import random
+from .. import data
+from tqdm import tqdm
+from ..data import dataset
+
+
+class TextClassificationDataset(data.Dataset):
```

Comment:
Is it possible to avoid creating this and have each dataset (e.g. AG_NEWS ) use only https://pytorch.org/docs/stable/data.html instead?

torchtext dataset abstractions need a lot of work and we should avoid the proliferation of torchtext-style datasets until we did a wider review.

This should be fine even if it means that we first release AG_NEWS, etc. with less features.